### PR TITLE
#142 Trusted Companies Component

### PR DESCRIPTION
**Pull request by alexfox361**
This PR attempts to solve issue #142. To do this, I added a new Trusted Companies Component. 

This component allows for the user to display a custom string and custom amount of images. 

I also added tests for this component. These test validate the input and rendering of the component's features. 

## Custom Fields
- Title
- Number of Companies
- Company Name
- Company Image

## Preview of Component 
image

## Files Changed

src/components/TrustedCompanies/TrustedCompanies.jsx:
- Added the design of the component. 

src/components/TrustedCompanies/TrustedCompanies.test.jsx:
- Tested the component for validation of input and rendering. 

src/components/TrustedCompanies/TrustedCompaniesComponent.js:
- Added the default options and rendering of the component. 

src/components/TrustedCompanies/TrustedCompaniesOptions.jsx:
- Added the customizable options of the component. 

src/components/TrustedCompanies/TrustedCompaniesPreview.png:
- Added a Preview of the component. 

src/utils/registeredComponents.js: 
- Added the component to the list of registered components.

### CODE REVIEW

**Review by CruzG-Dylan — CHANGES_REQUESTED** [review]
The component does seem to function as intended and the component does scale well with different screen sizes. Modifying the amount of companies using the add and delete buttons work, and the file upload and image urls change the component as intended. 

It does seem that the individual company titles aren't able to be changed even though there is an option to modify it in the tab. I would suggest either removing the option to change the title in general or updating it to match the intended use.

**Comment by alexfox361** [comment]
Thanks @CruzG-Dylan, I didn't notice that in testing. The bug should be fixed now.

**Review by CruzG-Dylan — APPROVED** [review]
Everything looks good now. Nice work.

**Review by ebacolod — CHANGES_REQUESTED** [review]
Testing, responsiveness, and inputs work well. However, one problem with the rendering is that when you add multiple components, the logos goe off the screen. 

image

Also, the 3rd company logo is missing some text. Zach has uploaded on the resources repo several logos that were used in the other Companies component. Here's the link: https://github.com/cs472-team6/cs472-team6-resources/tree/main/images/CompanyLogos. I believe the specific file is `DarkLogo4.svg`

**Comment by alexfox361** [comment]
Thanks @ebacolod, I made the changes and it should hopefully look better now.

File: src/components/TrustedCompanies/TrustedCompanies.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            </div>
+        </div>
+    )
+}
+
+function CompanyItem({Company}) {
+    const { Name, Picture } = Company;
+
+    return (
+        <img src={Picture} alt={Name} className="max-w-[130px] object-contain max-h-[130px] my-[20px] md:my-[60px] mr-[40px]"/>
```
Comment (ebacolod) [inline src/components/TrustedCompanies/TrustedCompanies.jsx:20]: When going in between screen sizes, the vertical margins between images are pretty large. 

image


I think if you lower the value of `md:my-[60px]`, it'll look much better. But also, this margin affects the spacing between other components, so make sure that it looks good in between components too.

**Review by ebacolod — CHANGES_REQUESTED** [review]

File: src/components/TrustedCompanies/TrustedCompanies.jsx (hunk 1 above)
Comment (alexfox361) [reply src/components/TrustedCompanies/TrustedCompanies.jsx]: Okay I lowered it to 20px. Let me know what you think.

**Review by ebacolod — APPROVED** [review]
LGTM!

**Pull request merged**